### PR TITLE
HHH-9857 - Reuse of EntityEntry for bytecode enhanced read-only refer…

### PR DESCRIPTION
**Pull request by johnaoahra80**
…ence cached entities

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/engine/internal/StatefulPersistenceContext.java [hunk 1]
```diff
@@ -489,13 +489,29 @@ public EntityEntry addEntry(
...
-			e.setStatus( status  );
+		/*
+		Important: 		the following instanceof checks and castings are intentional.
+						Please do *not* refactor to make calls through the EntityEntryFactory
+						interface. Refactoring will results in polymorphic call sites which will
+						severely impact performance.
+		*/
+		if (persister.getEntityEntryFactory() instanceof MutableEntityEntryFactory){
+			e = ((MutableEntityEntryFactory) persister.getEntityEntryFactory()).createEntityEntry(
+					status,
```
Comment (johnaoahra80) [inline hibernate-core/src/main/java/org/hibernate/engine/internal/StatefulPersistenceContext.java]: We are checking which EntityEntryfactory is referenced in the persister and casting to the implementation of the EntityEntry Factory. If we make the call via the interface, the call is to a polymorphic call site, which severely impacts performance, making the direct call means the JVM does not have to resolve which implementation to use.

**Comment by sebersole** [comment]
No, no, no!

You cannot reuse a Jira report that is closed against a released version to push more work.

**Comment by johnaoahra80** [comment]
I thought we were asked to reopen the existing jira for the performance issue. Shall I open a new jira?

**Comment by sebersole** [comment]
I said if the work is logically grouped with the work under an existing Jira that has not yet been released...

We have done some work for 5.0.4 (I don't have the Jira # in front of me) related to EntityEntry and perf.  And here is some more work related to EntityEntry and perf.  They are logically the same.

But you have to understand about reusing Jiras that are resolved against already released versions... That is how release notes are generated.  So you are essentially saying that this work is included in 5.0.0.CR2, which is clearly inaccurate.

**Pull request closed without merging**